qemu/tests/mq_change_qnum.py

```diff
@@ -29,30 +29,34 @@ def change_queues_number(session, ifname, q_number, queues_status=None):
         """
         Change queues number
         """
-        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         if not queues_status:
             queues_status = get_queues_status(session, ifname)
 
-        if (q_number != queues_status[1] and q_number <= queues_status[0] and
-                q_number > 0):
-            expect_status = 0
-        else:
-            expect_status = 1
-
+        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         status, output = session.cmd_status_output(mq_set_cmd, safe=True)
         cur_queues_status = get_queues_status(session, ifname)
-        if status != expect_status:
-            err_msg = "Change queues number failed, "
-            err_msg += "current queues set is %s, " % queues_status[1]
-            err_msg += "max allow queues set is %s, " % queues_status[0]
+
+        err_msg = ""
+        expect_q_number = q_number
+        if (q_number != queues_status[1] and q_number <= queues_status[0]
+                and q_number > 0):
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                err_msg = "Param is valid, but change queues failed, "
+        else:
+            if cur_queues_status != queues_status:
+                err_msg = "Param is invalid, no need change queues number, "
+                expect_q_number = queues_status[1]
```

Comment:
The above logic looks weird, you use 4 condition statements only for actually 2 conditions.